File: packages/web/src/api/overrideLog.ts
import api from "./index";

export interface OverrideLogItem {
  id: number;
  workorder_id: number;
  step_index: number;
  operator_badge: string;
  reason: string;
  video_url: string;
  created_at: string;
}

export interface OverrideLogListResult {
  items: OverrideLogItem[];
  total: number;
}

export interface OverrideStats {
  total: number;
  top_operators: { badge: string; count: number }[];
  daily_counts: { date: string; count: number }[];
}

export const overrideLogApi = {
  list: (params?: {
    workorder_id?: number;
    operator_badge?: string;
    start_date?: string;
    end_date?: string;
    skip?: number;
    limit?: number;
  }) => api.get<OverrideLogListResult>("/override-log/", { params }),
  create: (data: Omit<OverrideLogItem, "id" | "created_at">) =>
    api.post<OverrideLogItem>("/override-log/", data),
  get: (id: number) => api.get<OverrideLogItem>(`/override-log/${id}`),
  byWorkorder: (workorderId: number) =>
    api.get<OverrideLogItem[]>(`/override-log/workorder/${workorderId}`),
  stats: () => api.get<OverrideStats>("/override-log/stats"),
};

File: packages/server/src/schemas/override_log.py
from datetime import datetime

from pydantic import BaseModel, Field


class OverrideLogCreate(BaseModel):
    workorder_id: int
    step_index: int = Field(..., ge=0)
    operator_badge: str = Field(..., max_length=64)
    reason: str = ""
    video_url: str = ""


class OverrideLogResponse(BaseModel):
    id: int
    workorder_id: int
    step_index: int
    operator_badge: str
    reason: str
    video_url: str
    created_at: datetime

    model_config = {"from_attributes": True}


class TopOperatorItem(BaseModel):
    badge: str
    count: int


class DailyCountItem(BaseModel):
    date: str
    count: int


class OverrideStatsResponse(BaseModel):
    total: int
    top_operators: list[TopOperatorItem]
    daily_counts: list[DailyCountItem]


class OverrideLogListResponse(BaseModel):
    """分页列表（与前端表格分页配套）。"""

    items: list[OverrideLogResponse]
    total: int
